GitCommands/Statistics/ImpactLoader.cs

```diff
@@ -62,12 +70,20 @@ public ImpactLoader(IGitModule aModule)
 
         ~ImpactLoader()
         {
-            Stop();
+            Dispose(false);
         }
 
         public void Dispose()
         {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
             Stop();
```

Comment:
:exclamation: The call to `Stop` operates on managed resources, and therefore should only occur if `disposing` is true.